src/components/user/user.repository.ts

```diff
@@ -74,37 +75,54 @@ export class UserRepository extends DtoRepository(User) {
     ];
   }
 
-  private readonly roleProperties = (roles?: Role[]) =>
+  private readonly roleProperties = (roles?: Role[], baseNodeVar = 'user') =>
     (roles || []).flatMap((role) =>
-      property('roles', role, 'user', `role${role}`)
+      property('roles', role, baseNodeVar, `role${role}`)
     );
 
   async create(input: CreatePerson) {
     const id = await generateId();
-    const createdAt = DateTime.local();
+    const initialProps = {
+      realFirstName: input.realFirstName,
+      realLastName: input.realLastName,
+      displayFirstName: input.displayFirstName,
+      displayLastName: input.displayLastName,
+      phone: input.phone,
+      timezone: input.timezone,
+      about: input.about,
+      status: input.status,
+      title: input.title,
+      canDelete: true,
+    };
+    const baseNodeProps = { id };
```

Comment:
ID is generated inside of createNode and the DB query already returns it. So use that instead of generating it here.